cfme/common/vm.py

```diff
@@ -651,6 +657,143 @@ def does_vm_exist_on_provider(self):
         """Check if VM exists on provider itself"""
         return self.provider.mgmt.does_vm_exist(self.name)
 
+    def is_suspended_on_provider(self):
```

Comment:
OPTIONAL: honestly, I don't like that we have so much same on_appliance and on_provider methods. 
I'd consider either using sentaku or updating same methods in wrapanapi or creating one more class inside this class with on_provider methods like we did that in Appliance class.

Reply:
Well, I created all these methods because I had lambda assignments previously, and lint checks weren't liking that so I split them back out into methods ;) We could make these methods private, I could leave them defined inside the method itself, etc. But I figured it might be cleaner to just define them in the class.

Reply:
Or, instead of all the new defs we can go this route: https://gist.github.com/bsquizz/40ef2c5499bbd7b681f5b72dcd69c03e